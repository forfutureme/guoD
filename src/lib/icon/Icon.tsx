/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-09 14:05:56
 * @Desc: Icon 组件
 */
import React from 'react'
// 引入所有svg
import '@svg'
import './icon.scss'
import {prefixClass, joinClassName} from '@utils/classNameHandle'
const fixClass = prefixClass('guo-icon')

// 定义props 继承自React提供的SVG属性
interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string
}

const Icon: React.FC<IconProps> = ({name, className, ...restProps}) => {
  return (
    <svg className={joinClassName(fixClass(), className)} {...restProps}>
      <use xlinkHref={`#${name}`}></use>
    </svg>
  )
}

export default Icon
