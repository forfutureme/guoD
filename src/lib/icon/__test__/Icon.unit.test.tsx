/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-09 15:56:56
 * @Desc: icon 单元测试文件
 */
import renderer from 'react-test-renderer'
import {mount} from 'enzyme'
import React from 'react'
import Icon from '../Icon'

describe('Icon单元测试', () => {
  it('测试icon普通渲染', () => {
    const json = renderer.create(<Icon name="qq" />).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('测试icon的点击事件', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="qq" onClick={fn} />)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})
