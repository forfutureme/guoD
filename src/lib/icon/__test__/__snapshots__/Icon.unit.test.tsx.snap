// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon单元测试 测试icon普通渲染 1`] = `
<svg
  className="guo-icon guo-icon-qq"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;
